lib/gc/Analysis/MatmulConfigAnalysis.cpp

```diff
@@ -393,10 +400,15 @@ bool readConfigFromAttrs(MatmulConfig &config, ArrayRef<NamedAttribute> attrs) {
       cfgItemCnt++;
     }
   }
-  if (validateConfig(config)) {
-    return cfgItemCnt == 9;
-  } else {
-    LLVM_DEBUG(llvm::dbgs() << "The predefined config is invalid\n");
+  if (cfgItemCnt != 9) {
+    LLVM_DEBUG(llvm::dbgs() << "The predefined matmul config is incomplete. "
+                               "Default matmul config will be set.\n");
+    return false;
+  }
+  if (validateConfig(config, shape))
+    return true;
+  else {
+    assert(0 && "config is invalid");
```

Comment:
Better to use llvm::dbgs() to give the warning when there's invalid config, `assert` only throw error in debug mode.

Reply:
> Better to use llvm::dbgs() to give the warning when there's invalid config, `assert` only throw error in debug mode.

I will update to giving both `assert` and `llvm::dbgs()`.